articles/article-065-Information hiding.md

```diff
@@ -0,0 +1,36 @@
+<!-- BEGIN TITLE -->
+# Why are you hiding?
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+The idea for information hiding is to reduce complexity and to interact not with the information/data directly,
+but only indirectly via an interface. But why would we do this? Suppose we want to get a drink  from a vending machine,
+we need to select only, which drink we want and then pay for this, we never see the real work  and the logic behind the vending machine.
+This is the same for computer program, we use interfaces to hide  information/methods,
+this also means that we can interchange the hidden information easily as long as we do not change the interface.
+<!-- END BODY -->
+
+
+ 
+
+
+
+ 
+## Figure
+* A vending machine from the side, where we can see the interface(the button and a coin slit, and then the complexity of the inside)
+
+
+
+## Optional text
+<!-- BEGIN OPTIONAL -->
+* Here only the interface information hiding is mentioned and not  on the  packages  level( more user to program and not program to program).
```

Comment:
Please don't put review notes into the optional text section. Instead, add them as a comment to the pull request.